compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -1203,15 +1203,6 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                     err.emit()
                 } else if let Err(reported) = qself_ty.error_reported() {
                     reported
-                } else if let ty::Alias(ty::Opaque, alias_ty) = qself_ty.kind() {
-                    // `<impl Trait as OtherTrait>::Assoc` makes no sense.
-                    struct_span_code_err!(
-                        self.dcx(),
-                        tcx.def_span(alias_ty.def_id),
-                        E0667,
-                        "`impl Trait` is not allowed in path parameters"
-                    )
-                    .emit() // Already reported in an earlier stage.
```

Comment:
This should've been a `span_delayed_bug` in the first place (before this PR). This double emission (once during AST validation, once here during HIR ty lowering) actually leads to two duplicate diagnostics (indeed, even without `-Zdeduplicate-diagnostics=no`).